core/arch/arch.h

```diff
@@ -596,10 +596,10 @@ mangle_insert_clone_code(dcontext_t *dcontext, instrlist_t *ilist,
                          instr_t *instr _IF_X86_64(gencode_mode_t mode));
 
 #ifdef X86
-#    if defined(X64) || defined(MACOS)
+#    if defined(X64) || defined(UNIX)
+/* See i#847, i#3966 for discussion of stack alignment on 32-bit Linux. */
 #        define ABI_STACK_ALIGNMENT 16
 #    else
-/* See i#847 for discussing the stack alignment on X86. */
 #        define ABI_STACK_ALIGNMENT 4
```

Comment:
The Windows (MSVC-based) 32-bit ABI only mandates 4-byte alignment though so there isn't anything we have to fix for that -- though MinGW or Cygwin libraries built with gcc/clang will have 16-byte so I guess that's the complication.  But such libraries will have problems if they pass a callback to a Windows system library so maybe they're all asking for 4-byte alignment anyway?  A quick search finds stuff like https://github.com/uTox/uTox/issues/1304 which is apparently going to use -mstackrealign to solve this: which, if all MinGW/Cygwin builds do that, means we don't have to do anything.  It is worth filing an issue on this mixed-compiler-alignment problem => #4267.  I'll add a comment pointing to that.

Reply:
Ah I see. 

I took your PR message as the existence of pending work:

> Leaves Windows as 4-byte-aligned, since changing the injection
> assembly code and other pieces would be a bunch of extra work for
> little benefit: Windows 32-bit only requires 4-byte alignment, and the
> code is not that much more complex with a split alignment.

Reply:
I will try to reword that.  I did initially try to make Windows 16 too b/c I thought it would make DR's code simpler to have all platforms the same, but it ended up being more work so I reverted.